### PR TITLE
Feature/2024 02 23 performance testing

### PR DESCRIPTION
**Pull request by OumaimaZerouali**
### Thank you for your contribution! Please go through this small checklist to prepare your pull request to be reviewed as soon as possible :)
* [x] All authors were added to `_data/authors.yml` with a picture and an up-to-date bio
* [x] The blog post image is of ratio 500x293
  * Feel free to leave a comment if you don't know how to resize your image and someone will gladly help you out
* [x] The [**one sentence per line** method](https://raw.githubusercontent.com/brandon-rhodes/blog/master/texts/brandon/2012/one-sentence-per-line.rst) has been adopted
* [x] A spell check has been done
* [x] All links open in a new tab
  * Example: `[Amazing tech blog](https://ordina-jworks.github.io){:target="_blank" rel="noopener noreferrer"}`
* [x] The post's content properly resizes according to the window's size
  * Tip: Just shorten your window's horizontal length to easily test this
  * To have images properly resize use `class="image fit"`
  * To have YouTube videos properly resize, wrap the iframe into a `<div class="responsive-video">`
* [x] If a table of content is: check whether all links are working correctly

### Any specific remarks for the reviewers or anything you need help with?
\<enter text here>

### Checklist for the reviewers before merging
* [ ] Publishing date of the post has been correctly set
  * This can be done by renaming the filename or by setting the date in the header section of the post

### CODE REVIEW

File: _posts/2024-02-23-performance-testing.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+---
+layout: post
+authors: [oumaima-zerouali, viktor_van_steenweghen]
+title: 'The Imperfect Yet Crucial Role of Performance Testing'
+image: /img/2024-02-15-performance-testing/banner.png
```
Comment (nicholasM95) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
image: /img/2024-02-23-performance-testing/banner.png
```

File: _posts/2024-02-23-performance-testing.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+> Ah, the inner machinations of performance testing are an enigma.
+
+{:.no_toc}
+- TOC
+{:toc}
+
+## Introduction
+Attending FOSDEM 2024 was an eye-opening experience, especially the lecture on performance testing, which underscored its vital role in software development.
+
+It's incredible how something seemingly technical can have such a profound impact on the reliability, speed, and scalability of applications. In this article, we'll explore the critical role of performance testing in software development. We'll discuss its importance in identifying bottlenecks, enhancing user experience, and why it's essential for businesses. Additionally, we'll address the challenges of performance testing and introduce Gatling as a tool for database performance monitoring. Let's dive in!
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
It's incredible how something seemingly technical can profoundly impact the reliability, speed, and scalability of applications. In this article, we'll explore the critical role of performance testing in software development. We'll discuss its importance in identifying bottlenecks, enhancing user experience, and why it's essential for businesses. Additionally, we'll address the challenges of performance testing and introduce Gatling as a tool for database performance monitoring. Let's dive in!
```

File: _posts/2024-02-23-performance-testing.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+## Why Performance Testing Matters
+Performance testing serves as a crucial element in guaranteeing that applications not only meet but surpass the expectations of both users and businesses. It transcends basic functionality testing, delving deeply into the nuances of how an application operates under diverse conditions and workloads.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
Performance testing is crucial in guaranteeing that applications not only meet but surpass the expectations of both users and businesses. It transcends basic functionality testing, delving deeply into the nuances of how an application operates under diverse conditions and workloads.
```

File: _posts/2024-02-23-performance-testing.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+### Identifying and Enhancing Performance Bottlenecks to Improve User Experience
+One of the primary objectives of performance testing is to identify bottlenecks within an application. These bottlenecks could manifest as slow response times, excessive resource consumption, or scalability limitations. By pinpointing these bottlenecks early in the development cycle, developers can take proactive measures to address them, thereby optimizing the overall performance of the application.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
One of the primary objectives of performance testing is to identify bottlenecks within an application. These bottlenecks could manifest as slow response times, excessive resource consumption, or scalability limitations. By pinpointing these bottlenecks early in the development cycle, developers can take proactive measures to address them, thereby optimizing the application's overall performance.
```

File: _posts/2024-02-23-performance-testing.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+In today's digital landscape, user experience reigns supreme. Poor performance can lead to frustration, dissatisfaction, and ultimately, abandonment of the application. Performance testing plays a crucial role in ensuring that the application delivers a seamless and responsive user experience, regardless of the user's device, location, or network conditions. By optimizing performance, businesses can enhance user satisfaction, loyalty, and retention.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
In today's digital landscape, user experience reigns supreme. Poor performance can lead to frustration, dissatisfaction, and, ultimately, abandonment of the application. Performance testing ensures the application delivers a seamless and responsive user experience, regardless of the user's device, location, or network conditions. Businesses can enhance user satisfaction, loyalty, and retention by optimizing performance.
```

File: _posts/2024-02-23-performance-testing.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+### Compelling Reasons to Prioritize Performance Testing
+Here are several compelling reasons for prioritizing performance testing:
+- **Performance Gain:** Optimizing product performance directly translates to tangible benefits for businesses, including increased customer satisfaction, higher conversion rates, and improved bottom-line revenue.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
- **Performance Gain:** Optimizing product performance directly translates to tangible business benefits, including increased customer satisfaction, higher conversion rates, and improved bottom-line revenue.
```

File: _posts/2024-02-23-performance-testing.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+- **Deeper Understanding:** Performance testing provides developers with invaluable insights into the underlying intricacies of an application's architecture, codebase, and infrastructure. By understanding how the application performs under different conditions, developers can make informed decisions to optimize its performance.
+- **Baseline Figures:** Obtaining baseline performance metrics serves as a foundational reference point for future iterations of the application. These metrics provide a benchmark against which future performance improvements can be measured, enabling continuous enhancement and refinement of the application.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
- **Baseline Figures:** Obtaining baseline performance metrics serves as a foundational reference point for measuring future performance improvements, enabling the continuous enhancement and refinement of the application. These metrics establish a benchmark for comparing future improvements and facilitating ongoing development.
```

Geschreven in een actievere vorm.

File: _posts/2024-02-23-performance-testing.md [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+In essence, performance testing is not just a technical necessity but a strategic imperative for businesses looking to deliver high-quality, high-performing applications that delight users and drive success in today's hyper-competitive digital landscape.
+
+## Challenges and the Imperfect Nature of Performance Testing
+Acknowledging the imperfections inherent in performance testing is crucial for understanding the challenges faced in this domain. Despite these challenges, it's imperative to recognize the importance of imperfect performance testing and how it continues to provide value in software development:
+
+### Early Detection and Continuous Improvement
+Even in its imperfect state, performance testing serves as a vital tool for early issue detection and continuous improvement. By uncovering critical issues and bottlenecks early in the development cycle, developers can take corrective action before they escalate into more significant problems. This iterative refinement process fosters a culture of continuous improvement, leading to incremental performance gains over time.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
Even in its imperfect state, performance testing is vital for early issue detection and continuous improvement. By uncovering critical issues and bottlenecks early in the development cycle, developers can take corrective action before they escalate into more significant problems. This iterative refinement process fosters a culture of continuous improvement, leading to incremental performance gains over time.
```

File: _posts/2024-02-23-performance-testing.md [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+
+### Real-World Insights and Complexity
+Imperfect performance testing provides invaluable insights into how applications behave under real-world conditions, including stress, resource limitations, and unexpected usage patterns. However, testing such complex systems requires a deep understanding of their inner workings, posing a significant challenge. Additionally, the dynamic nature of software environments introduces further complexity, making it impossible to replicate every possible scenario in controlled testing environments.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
Imperfect performance testing provides invaluable insights into how applications behave under real-world conditions, including stress, resource limitations, and unexpected usage patterns. However, testing such complex systems requires a deep understanding of their inner workings, posing a significant challenge. Additionally, the dynamic nature of software environments introduces further complexity, making replicating every possible scenario in controlled testing environments impossible.
```

File: _posts/2024-02-23-performance-testing.md [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+
+##   Essential Components for Effective Performance Testing
+When undertaking performance testing, it's vital to consider various components to ensure a thorough evaluation:
+
+### Load Testing & Stress Testing
+Load testing involves simulating real-world loads on web services to assess their performance under normal conditions, while stress testing pushes the system to its limits to identify its breaking point. Tools like Gatling offer scripting capabilities to define complex load scenarios and analyze performance under varying loads. Load testing helps gauge how the application handles typical user loads, while stress testing identifies weaknesses and potential failure points under extreme conditions.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md:55]:
```suggestion
Load testing involves simulating real-world loads on web services to assess their performance under normal conditions. In contrast, stress testing pushes the system to its limits to identify its breaking point. Tools like Gatling offer scripting capabilities to define complex load scenarios and analyze performance under varying loads. Load testing helps gauge how the application handles typical user loads, while stress testing identifies weaknesses and potential failure points under extreme conditions.
```

File: _posts/2024-02-23-performance-testing.md [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+
+### Endurance Testing
+Endurance testing evaluates how well the application performs over prolonged periods, ensuring stability without memory leaks or performance degradation. Continuous load tests running for several hours or days monitor metrics like memory usage, CPU utilization, and other system parameters. Endurance testing is crucial for identifying any degradation in performance over time, ensuring that the application remains stable and reliable during extended usage.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
Endurance testing evaluates the application's performance over prolonged periods, ensuring stability without memory leaks or performance degradation. Continuous load tests running for several hours or days monitor metrics like memory usage, CPU utilization, and other system parameters. Endurance testing is crucial for identifying any degradation in performance over time, ensuring that the application remains stable and reliable during extended usage.
```

File: _posts/2024-02-23-performance-testing.md [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+
+### Scalability Testing
+Scalability testing assesses how well the application can handle increased load by measuring metrics like throughput, response time, and resource utilization. Tools like Gatling can simulate gradual increases in the number of users or transactions to observe how the system scales horizontally or vertically. Scalability testing helps identify performance bottlenecks and capacity limits before deploying the application to production, ensuring that it can accommodate growth without sacrificing performance.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md:61]:
```suggestion
Scalability testing assesses how well the application can handle increased load by measuring throughput, response time, and resource utilization metrics. Tools like Gatling can simulate gradual increases in the number of users or transactions to observe how the system scales horizontally or vertically. Scalability testing helps identify performance bottlenecks and capacity limits before deploying the application to production, ensuring that it can accommodate growth without sacrificing performance.
```

File: _posts/2024-02-23-performance-testing.md [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+
+### API Performance
+APIs play a critical role in modern web applications, facilitating communication between different components. API performance testing, conducted using tools like Apache JMeter, involves sending HTTP requests and measuring response times to ensure that APIs meet performance requirements. Simulating various API calls with different payloads helps analyze how backend services respond under different load conditions. API performance testing ensures that APIs can handle expected traffic volumes without degradation, maintaining optimal performance for end users.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
APIs are critical in modern web applications, facilitating communication between different components. API performance testing using tools like Apache JMeter involves sending HTTP requests and measuring response times to ensure that APIs meet performance requirements. Simulating various API calls with different payloads helps analyze how backend services respond under different load conditions. API performance testing ensures that APIs can handle expected traffic volumes without degradation, maintaining optimal performance for end users.
```

File: _posts/2024-02-23-performance-testing.md [hunk 14]
```diff
@@ -0,0 +1,128 @@
...
+
+### Automation & Integration
+With the adoption of DevOps practices, automation and integration are essential for streamlining the performance testing process. Tools like Apache Maven enable the automation of performance tests as part of the continuous integration pipeline. Integrating performance testing tools with build servers like Jenkins or CI/CD platforms like GitHub ensures that performance tests are run automatically on code changes. Additionally, open-source automation frameworks like Selenium can be used to automate web application tests, complementing performance testing efforts and ensuring comprehensive test coverage across the application stack. Automation and integration help improve efficiency, consistency, and reliability in performance testing processes, enabling faster feedback and quicker resolution of performance issues.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
With the adoption of DevOps practices, automation, and integration are essential for streamlining the performance testing process. Tools like Apache Maven enable the automation of performance tests as part of the continuous integration pipeline. Integrating performance testing tools with build servers like Jenkins or CI/CD platforms like GitHub ensures that performance tests are run automatically on code changes. Additionally, developers can use open-source automation frameworks like Selenium to automate web application tests, complementing performance testing efforts and ensuring comprehensive test coverage across the application stack. Automation and integration help improve efficiency, consistency, and reliability in performance testing processes, enabling faster feedback and quicker resolution of performance issues.
```

File: _posts/2024-02-23-performance-testing.md [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+
+## Performance testing with Gatling
+Gatling is a powerful open-source tool designed for performance testing, renowned for its efficiency and flexibility. It allows developers to simulate real-world scenarios and assess the performance of their applications under various load conditions. Gatling uses a scenario-based approach, where users define test scenarios using a simple yet expressive DSL (Domain-Specific Language). These scenarios can simulate user interactions, such as browsing web pages, submitting forms, or making API calls.
+
+### Gatling
+During our devcase, we came into contact with Gatling for the first time. The task was to build an application that could retrieve data from a smart electricity meter and translate it into something understandable for people. So, data had to be stored and processed. For this, we used a lambda to convert the raw data and store it in a Timestream database.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
During our devcase, we came into contact with Gatling for the first time. The task was to build an application that could retrieve data from an intelligent electricity meter and translate it into something understandable for people. So, data had to be stored and processed. We used lambda to convert the raw data and store it in a Timestream database.
```

File: _posts/2024-02-23-performance-testing.md [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+
+To avoid the application crashing when more than 5 users simultaneously send the data from their meter, it was important to perform performance testing. It was important to know approximately how many simultaneous users the system could handle before it failed. For this, we used Gatling.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
It was essential to perform performance testing to avoid the application crashing when more than five users simultaneously send the data from their meter. It was important to know approximately how many simultaneous users the system could handle before it failed. For this, we used Gatling.
```

File: _posts/2024-02-23-performance-testing.md [hunk 17]
```diff
@@ -0,0 +1,128 @@
...
+### Structure of Gatling Code
+The Gatling code is structured as follows:
+
+#### The httpProtocolBuilder
+The httpProtocolBuilder is a fundamental component in Gatling scripts, responsible for defining the configuration of the HTTP protocol used in our performance tests. It allows us to set various parameters and characteristics related to HTTP communication, ensuring accurate simulation of real-world scenarios.
+
+The configuration options provided by the `HTTPProtocolBuilder` enable precise control over aspects such as:
+- Request Headers: Specification of headers such as User-Agent, Content-Type, and Accept headers, which mimic the   behavior of actual web browsers or API clients.
+- Request Timeouts: Setting timeouts for establishing connections, receiving responses, or overall request durations, ensuring realistic behavior under varying network conditions.
+- Proxy Configuration: Configuring proxy settings for simulating requests through intermediary servers, relevant for testing applications deployed behind proxies.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
- Proxy Configuration: Configuring proxy settings for simulating requests through intermediary servers relevant for testing applications deployed behind proxies.
```

File: _posts/2024-02-23-performance-testing.md [hunk 18]
```diff
@@ -0,0 +1,128 @@
...
+- SSL/TLS Settings: Configuring SSL/TLS parameters such as cipher suites, SSL protocols, and certificate validation settings to mimic secure communication.
+- Response Handling: Defining how Gatling should handle responses, including following redirects, handling cookies, and managing session data.
+
+By customizing these settings within the httpProtocolBuilder, we can create test scenarios that closely resemble the behavior of real users interacting with our application or API.
+
+![httpProtocol.png](..%2Fimg%2F2024-02-23-performance-testing%2FhttpProtocol.png)
+
+#### The ScenarioBuilder
+In Gatling, the ScenarioBuilder plays a pivotal role in crafting realistic user interactions with the web application being tested. It serves as the blueprint for defining various user journeys or workflows, encompassing the sequence of HTTP requests to be made and the order in which they should occur.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
In Gatling, the ScenarioBuilder crucially crafts realistic user interactions with the web application under test. It acts as the blueprint for defining various user journeys or workflows, detailing the sequence of HTTP requests to make and the order they should follow.
```

File: _posts/2024-02-23-performance-testing.md [hunk 19]
```diff
@@ -0,0 +1,128 @@
...
+
+A `ScenarioBuilder` typically includes the following key elements:
+- HTTP Requests: Specification of the HTTP requests to be sent to the target application or API. These requests represent user actions such as navigating to a webpage, submitting a form, or interacting with API endpoints.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
- HTTP Requests: Developers specify the HTTP requests to send to the target application or API, representing user actions such as navigating a webpage, submitting a form, or interacting with API endpoints.
```

File: _posts/2024-02-23-performance-testing.md [hunk 20]
```diff
@@ -0,0 +1,128 @@
...
+- Order of Execution: Determination of the sequence in which the HTTP requests should be executed within the scenario. This sequencing ensures that the simulated user interactions align with real-world usage patterns.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
- Order of Execution: They determine the sequence to execute the HTTP requests within the scenario, ensuring the simulated user interactions align with real-world usage patterns.
```

File: _posts/2024-02-23-performance-testing.md [hunk 21]
```diff
@@ -0,0 +1,128 @@
...
+- Think Times: Introduction of delays or "think times" between requests to mimic the pauses users naturally make while interacting with the application. These pauses add realism to the simulation and help avoid overloading the server with rapid-fire requests.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
- Think Times: Developers introduce delays or "think times" between requests to mimic users' natural pauses while interacting with the application. These pauses add realism to the simulation and help avoid overloading the server with rapid-fire requests.
```

File: _posts/2024-02-23-performance-testing.md [hunk 22]
```diff
@@ -0,0 +1,128 @@
...
+
+The ScenarioBuilder empowers us to create diverse and intricate scenarios that mirror the complexity of user interactions in production environments. By accurately replicating user behavior, we can assess how the application performs under different usage scenarios and identify any performance bottlenecks or issues.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
The ScenarioBuilder enables the creation of diverse and intricate scenarios that mirror the complexity of user interactions in production environments. Accurately replicating user behavior allows for assessing application performance under different usage scenarios and identifying any performance bottlenecks or issues.
```

File: _posts/2024-02-23-performance-testing.md [hunk 23]
```diff
@@ -0,0 +1,128 @@
...
+
+![scenario.png](..%2Fimg%2F2024-02-23-performance-testing%2Fscenario.png)
+
+#### The Setup
+This is a method in Gatling that allows you to set up the test scenarios and configure the simulation before it is executed. It accepts one or more scenarios and executes them. This consists of 2 parts:
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
This method in Gatling lets you set up the test scenarios and configure the simulation before executing it. It accepts one or more scenarios and executes them. A scenario consists of 2 parts:
```

File: _posts/2024-02-23-performance-testing.md [hunk 24]
```diff
@@ -0,0 +1,128 @@
...
+
+**scn.injectOpen(...)**: This segment of the code is responsible for configuring the injection of user behavior into the scenario (scn). Here, we define the pattern or strategy for simulating user interactions within our test scenarios. For instance, we can specify how users are injected into the scenario over time, whether it's a gradual ramp-up, a constant load, or a spike in user activity.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md:109]:
```suggestion
**scn.injectOpen(...)**: This code segment configures the injection of user behavior into the scenario (scn). Here, we define the pattern or strategy for simulating user interactions within our test scenarios. For instance, we can specify how users are injected into the scenario over time, whether it's a gradual ramp-up, a constant load, or a spike in user activity.
```

File: _posts/2024-02-23-performance-testing.md [hunk 25]
```diff
@@ -0,0 +1,128 @@
...
+
+**rampUsers(100).during(100)**: This part of the setup specifies the injection pattern for users. In this example, it indicates that the number of virtual users will gradually increase from 0 to 100 over a duration of 100 seconds. In simpler terms, with each passing second, Gatling will introduce an additional virtual user into the scenario until reaching a total of 100 concurrent users. This gradual ramp-up helps simulate realistic user load patterns and allows us to observe how the system performs under increasing levels of stress.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
**rampUsers(100).during(100)**: This part of the setup specifies the injection pattern for users. This example indicates that the number of virtual users will gradually increase from 0 to 100 for 100 seconds. In simpler terms, with each passing second, Gatling will introduce an additional virtual user into the scenario until reaching a total of 100 concurrent users. This gradual ramp-up helps simulate realistic user load patterns and allows us to observe how the system performs under increasing stress levels.
```

File: _posts/2024-02-23-performance-testing.md [hunk 26]
```diff
@@ -0,0 +1,128 @@
...
+
+![setup.png](..%2Fimg%2F2024-02-23-performance-testing%2Fsetup.png)
+
+#### Results
+Following the completion of Gatling tests, a comprehensive report is generated, providing valuable insights into the performance of the tested application. Here are some key observations typically found in the report:
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md:116]:
```suggestion
Following the completion of Gatling tests, Gatling generates a comprehensive report, providing valuable insights into the performance of the tested application. Here are some key observations typically found in the report:
```

File: _posts/2024-02-23-performance-testing.md [hunk 27]
```diff
@@ -0,0 +1,128 @@
...
+
+**Response Times**: The report includes data on various metrics related to response times, such as average, minimum, maximum, and 95th percentile response times. These metrics indicate how quickly the application responds to different types of requests. Lower response times are generally preferable as they signify faster application responsiveness.
+![response.png](..%2Fimg%2F2024-02-23-performance-testing%2Fresponse.png)
+
+**Errors**: Information regarding any errors encountered during the test is documented in the report. These errors may include server errors, timeouts, or incorrect responses. Identifying and addressing these errors is crucial for improving application reliability and user experience.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
**Errors**: The report documents information regarding any errors encountered during the test. These errors may include server errors, timeouts, or incorrect responses. Identifying and addressing these errors is crucial for improving application reliability and user experience. 
```

File: _posts/2024-02-23-performance-testing.md [hunk 28]
```diff
@@ -0,0 +1,128 @@
...
+![fouten.png](..%2Fimg%2F2024-02-23-performance-testing%2Ffouten.png)
+
+From the results obtained, we can infer that the tested lambda function and database are capable of handling requests from 100 concurrent users. However, it is noted that the average response time for such requests is approximately 1200 ms. While functional, this response time may be considered relatively long and suggests room for optimization to enhance overall system performance.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
Based on the results, we infer that the tested lambda function and database can handle requests from 100 concurrent users. However, we note that the average response time for these requests is around 1200 ms. Although the system functions, this response time is relatively long, indicating potential for optimization to improve overall system performance.
```

File: _posts/2024-02-23-performance-testing.md [hunk 29]
```diff
@@ -0,0 +1,128 @@
...
+## Conclusion
+In the realm of software development, performance testing serves as the backbone, ensuring applications stand strong in terms of reliability, scalability, and efficiency. While the pursuit of perfection may seem daunting, the quirks and challenges inherent in the process remind us of the complex nature of modern software and the evolving expectations of users.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
In software development, performance testing is the backbone, ensuring applications stand firm regarding reliability, scalability, and efficiency. While the pursuit of perfection may seem daunting, the quirks and challenges in the process remind us of the complex nature of modern software and users' evolving expectations.
```

File: _posts/2024-02-23-performance-testing.md [hunk 30]
```diff
@@ -0,0 +1,128 @@
...
+This analysis highlights the critical importance of performance testing. It was observed that while the application could handle the requests, it still took a long time to process them. This underscores the significance of optimizing performance. Addressing these identified issues and implementing necessary improvements can enhance application efficiency, scalability, and security, ultimately delivering a superior user experience.
```
Comment (denmette) [inline _posts/2024-02-23-performance-testing.md]:
```suggestion
This analysis highlights the critical importance of performance testing. Observers noted that, although the application could handle the requests, processing them took a long time. This underscores the significance of optimizing performance. Addressing these issues and implementing necessary improvements can enhance application efficiency, scalability, and security, ultimately delivering a superior user experience.
```

**Review by denmette — CHANGES_REQUESTED** [review]
@ViktorVS1 en @OumaimaZerouali 

Mooi geschreven, bedankt kleine zaken die ik anders zou schrijven. Een actieve vorm van schrijven is leuker voor de lezer (of alvast zo ervaar ik het).

Ik denk dat jullie wel nog eens best de website lokaal op starten via docker, zodat je de layout ook nog even kunnen bekijken.

**Review by denmette — APPROVED** [review]
